src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -377,7 +383,21 @@ public boolean equals(Object o)
         return Objects.equals(keyspaceName, snapshot.keyspaceName) &&
                Objects.equals(tableName, snapshot.tableName) &&
                Objects.equals(tableId, snapshot.tableId) &&
-               Objects.equals(tag, snapshot.tag);
+               tagsEqual(tag, snapshot.tag);
+    }
+
+    private boolean tagsEqual(String tag1, String tag2)
+    {
+        if (tag1 == null && tag2 == null)
+            return true;
+
+        if (tag1 == null || tag2 == null)
+            return false;
+
+        if (PLATFORM == PlatformEnum.MACOS)
+            return LocalizeString.toLowerCaseLocalized(tag1).equals(LocalizeString.toLowerCaseLocalized(tag2));
```

Comment:
Nit: I think this is one of those places that would benefit from a comment stating why we treat Mac differently.